### DISCUSSION TITLE
number of connected messages

### DISCUSSION THREAD

**Discussion by crasty01** [supabase/supabase#2276] [Questions] [answered]
I have 2 tables, one for users, one for messages which are linked to a specific user by user-id.
In my js frontend, I have a simple request:
```js
const res = await supabase.from('users').select(`display-name`).ilike('display-name', `%${value}%`);
```
which searches the display-names for a match.

I would like for the request to return a display-names that match AND a number of messages connected to the user's user-id (number of messages sent by the user),
so that data is something like
```js
[
 {'display-name': 'name01', count: 13},
 {'display-name': 'name02', count: 138},
 {'display-name': 'name03', count: 57}
]
```
Is that possible in one request or do I need to make 2?

*Thanks beforehand to anyone answering.*

**Comment by steve-chavez** [comment, marked as answer]
Hey @crasty01,

Try doing:

```js
await supabase
  .from('users')
  .select(`display-name,messages(count)`)
  .ilike('display-name', `%${value}%`);
```

You should get a shape like:

```js
[
 {'display-name': 'name01', messages:[{count: 13}]},
 {'display-name': 'name02', messages:[{count: 138}]}
]
```

**Reply by crasty01** [reply]
Hey, I just tried it and it works, thanks.

Unfortunately, I won't be able to use it. The main idea was for the autocomplete to sort out the most important users, those with the most messages sent, and give them as a suggestion and the count takes way too long for this use.

This is my first time really working with a database, so at least I learned something.
Or maybe there is a way to estimate the count to make it faster?

Either way, thanks a lot for answering 😄

**Reply by steve-chavez** [reply]
@crasty01 If you have a large number of rows, you can use the `estimated` count option, see [here](https://supabase.io/docs/reference/javascript/select#querying-with-count-option).
